cartography/graph/cleanupbuilder.py

```diff
@@ -0,0 +1,290 @@
+        return _build_cleanup_node_sub_resource_only(
+            node_schema,
+            sub_resource_link,
+            node_schema.sub_resource_relationship.target_node_matcher,
+        )
+
+    # Draw selected relationship with correct direction
+    if selected_relationship.direction == LinkDirection.INWARD:
+        selected_rel_template = Template("<-[:$SelectedRelLabel]-")
+    else:
+        selected_rel_template = Template("-[:$SelectedRelLabel]->")
+    selected_rel = selected_rel_template.safe_substitute(SelectedRelLabel=selected_relationship.rel_label)
+
+    # Ensure the node is attached to the sub resource and delete the node
+    query_template = Template(
+        """
+        MATCH (n:$node_label)$sub_resource_link(:$sub_resource_label{$match_sub_res_clause})
+        MATCH (n)$selected_rel(:$other_node_label)
+        WHERE n.lastupdated <> $UPDATE_TAG
+        WITH n LIMIT $LIMIT_SIZE
+        DETACH DELETE n;
+        """,
+    )
+    return query_template.safe_substitute(
+        node_label=node_schema.label,
+        sub_resource_link=sub_resource_link,
+        sub_resource_label=node_schema.sub_resource_relationship.target_node_label,
+        match_sub_res_clause=_build_match_clause(node_schema.sub_resource_relationship.target_node_matcher),
+        selected_rel=selected_rel,
+        other_node_label=selected_relationship.target_node_label,
+    )
+
+
+def _build_cleanup_node_sub_resource_only(
+        node_schema: CartographyNodeSchema,
+        sub_res_link: str,
+        sub_res_node_matcher: TargetNodeMatcher,
+) -> str:
+    """
+    Generate a query to clean up a node attached to a sub resource.
+    :param node_schema: The CartographyNodeSchema object
+    :param sub_res_link: A string that looks like "<-[:RELATIONSHIP]-" or "-[:RELATIONSHIP]->"
+    :param sub_res_node_matcher: The TargetNodeMatcher object used to determine which sub resource to query for when
+    cleaning up.
+    """
+    query_template = Template(
+        """
+        MATCH (n:$node_label)$sub_res_link(:$sub_resource_label{$sub_res_match_clause})
+        WHERE n.lastupdated <> $UPDATE_TAG
+        WITH n LIMIT $LIMIT_SIZE
+        DETACH DELETE n;
+        """,
+    )
+    if not node_schema.sub_resource_relationship:
+        raise ValueError(
+            f"'_build_cleanup_node_sub_resource_only() failed: {node_schema.label}' does not have a "
+            "sub_resource_relationship defined, so we cannot generate a query to clean it up. Please verify that the "
+            "class definition is what you expect.",
+        )
+    return query_template.safe_substitute(
+        node_label=node_schema.label,
+        sub_resource_label=node_schema.sub_resource_relationship.target_node_label,
+        sub_res_link=sub_res_link,
+        sub_res_match_clause=_build_match_clause(sub_res_node_matcher),
+    )
+
+
+def _build_cleanup_rel_query(
```

Comment:
`_build_cleanup_node_query()` and `_build_cleanup_rel_query()` can be combined into a `build_node_and_rel_cleanup_queries()`, since we would always do them in pairs. They could just return the two as `(node_cleanup_query, rel_cleanup_query)`